test/metabase/api/persist_test.clj

```diff
@@ -43,3 +45,39 @@
         (is (= default-cron
                (get-in (task.persist-refresh/job-info-by-db-id)
                        [(:id db) :schedule])))))))
+
+(deftest persisted-info-by-id-test
+  (with-setup db
+    (t2.with-temp/with-temp
+      [:model/Card          model     {:database_id (u/the-id db), :dataset true}
+       :model/PersistedInfo pmodel    {:database_id (u/the-id db), :card_id (u/the-id model)}]
+      (testing "Should require a non-negative persisted-info-id"
+        (is (= "API endpoint does not exist."
+               (mt/user-http-request :crowberto :get 404 (format "persist/%d" -1)))))
+      (testing "Should not get info when the persisted-info-id doesn't exist"
+        (is (= "Not found."
+               (mt/user-http-request :crowberto :get 404 (format "persist/%d" (+ 1 (u/the-id pmodel)))))))
```

Comment:
```suggestion
               (mt/user-http-request :crowberto :get 404 (format "persist/%d" Integer/MAX_VALUE)))))
```

simply increase does not guarantee it exists. Tests could be run in parallel so it's possible to have other process creating persisted info

Reply:
Good point, it could exist from another test. 

Committed your suggestion in https://github.com/metabase/metabase/pull/33776/commits/d90c9c554af0c32960850f59856fb28fe9974bd8.